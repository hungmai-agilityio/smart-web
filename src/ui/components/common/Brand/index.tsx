import Link from 'next/link';
import Image from 'next/image';

interface BrandProps {
  logo: string;
  url?: string;
}

export const Brand = ({ logo, url = '/' }: BrandProps) => {
  return (
    <Link href={url} className="cursor-pointer">
      <Image src={logo} alt="logo" width={160} height={84} priority />
    </Link>
  );
};
