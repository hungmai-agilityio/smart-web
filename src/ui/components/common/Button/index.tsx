'use client';

import { ButtonHTMLAttributes } from 'react';
import { clsx } from 'clsx';

// Constants
import { TYPE } from '@/constants';

interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  variant?: TYPE.PRIMARY | TYPE.SECOND;
  styles?: string;
}

export const Button = ({ name, variant, styles, disabled }: ButtonProps) => {
  const typeClasses = {
    [TYPE.PRIMARY]: 'bg-second',
    [TYPE.SECOND]:
      'bg-white border-primary border-2 border-primary hover:bg-second hover:border-0'
  };

  const typeClass = typeClasses[variant || TYPE.PRIMARY];

  const handleClick = () => {
    alert('We will update feature later');
  };

  return (
    <button
      className={clsx(
        'text-primary px-2 rounded-full font-light truncate',
        'w-24 h-6 text-xs sm:w-28 sm:h-8 sm:text-sm md:w-36 md:h-10 md:text-base lg:w-52 lg:h-12 lg:text-lg',
       disabled && 'cursor-not-allowed hover:bg-gray-400',
        typeClass,
        styles
      )}
      onClick={handleClick}
      disabled={disabled}
    >
      {name}
    </button>
  );
};
