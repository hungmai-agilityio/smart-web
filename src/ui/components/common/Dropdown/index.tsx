'use client';
import { useRef, useState } from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCaretDown } from '@fortawesome/free-solid-svg-icons';
import { useOutsideClick } from '@/hooks/useOutsideClick';

export interface ItemProps {
  label: string;
  value: string;
}

interface DropdownProps {
  items: ItemProps[];
  onSelect?: (value: string) => void;
}

export const Dropdown = ({ items, onSelect }: DropdownProps) => {
  const [isOpen, setIsOpen] = useState(false);
  const [selectedItem, setSelectedItem] = useState<ItemProps>(items[0]);

  const dropdownRef = useRef<HTMLDivElement>(null);

  const toggleDropdown = () => {
    setIsOpen((prev) => !prev);
  };

  const handleItemClick = (item: ItemProps) => {
    setSelectedItem(item);
    setIsOpen(false);
    if (onSelect) onSelect(item.value);
  };

  useOutsideClick({
    ref: dropdownRef,
    handler: () => setIsOpen(false)
  });

  return (
    <div ref={dropdownRef} className="relative inline-block text-left w-36">
      <button
        type="button"
        className="flex items-center justify-between w-full px-3 py-2 border border-primary rounded-full md:text-base text-x-xs uppercase"
        onClick={toggleDropdown}
      >
        {selectedItem.label}
        <FontAwesomeIcon
          icon={faCaretDown}
          className={`ml-2 transition-transform ${isOpen ? 'rotate-180' : 'rotate-0'}`}
        />
      </button>

      <div
        className={`absolute right-0 w-full mt-1 bg-white border border-gray-300 rounded-md shadow-lg z-10 transition-transform duration-300 ease-in-out ${
          isOpen ? 'opacity-100 translate-y-0' : 'opacity-0 translate-y-2'
        }`}
      >
        {items.map((item) => (
          <button
            key={item.value}
            className="w-full text-left px-4 py-2 hover:bg-gray-100 md:text-base text-xs uppercase"
            onClick={handleItemClick.bind(null, item)}
          >
            {item.label}
          </button>
        ))}
      </div>
    </div>
  );
};
