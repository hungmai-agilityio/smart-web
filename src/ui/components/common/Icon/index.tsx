import { clsx } from 'clsx';

// Font Awesome
import { IconDefinition } from '@fortawesome/fontawesome-svg-core';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

interface IconProp {
  icon: IconDefinition;
  quantity?: number;
  styles?: string;
}
export const Icon = ({ icon, quantity, styles }: IconProp) => {
  return (
    <div className={clsx('flex gap-3 items-center', styles)}>
      <FontAwesomeIcon icon={icon} />

      <p>{quantity}</p>
    </div>
  );
};
