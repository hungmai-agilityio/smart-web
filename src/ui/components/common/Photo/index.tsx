'use client';
import { ReactNode, useState } from 'react';
import { clsx } from 'clsx';
import Image, { ImageProps } from 'next/image';

interface PhotoProps extends ImageProps {
  children?: ReactNode;
  styles?: string;
  isCircle?: boolean;
  fixedSize?: { width: number; height: number };
  autoSize?: boolean;
}

export const Photo = ({
  src,
  alt,
  children,
  styles,
  isCircle,
  fixedSize,
  autoSize,
  ...res
}: PhotoProps) => {
  const [hovered, setHovered] = useState(false);

  const toggleHover = () => setHovered(!hovered);

  const containerStyles = clsx('relative', styles, {
    'rounded-full overflow-hidden border-4 border-gray-300 hover:border-second':
      isCircle,
    'w-full h-full': autoSize,
    [`w-[${fixedSize?.width}px] h-[${fixedSize?.height}px]`]:
      fixedSize && !autoSize
  });

  return (
    <div
      className={containerStyles}
      onMouseEnter={toggleHover}
      onMouseLeave={toggleHover}
    >
      <Image
        src={src}
        alt={alt}
        fill={autoSize}
        width={fixedSize?.width}
        height={fixedSize?.height}
        style={{ objectFit: 'cover' }}
        className={clsx('transition-opacity duration-300', {
          'opacity-50': hovered && children
        })}
        {...res}
      />
      {hovered && children && (
        <div className="absolute inset-0 flex items-center bg-black bg-opacity-50 text-white">
          {children}
        </div>
      )}
    </div>
  );
};
