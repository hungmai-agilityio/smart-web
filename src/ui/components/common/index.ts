export * from '@/ui/components/common/Button';
export * from '@/ui/components/common/Dropdown';
export * from '@/ui/components/common/Icon';
export * from '@/ui/components/common/Photo';
export * from '@/ui/components/common/Text';
export * from '@/ui/components/common/StatsRow';
