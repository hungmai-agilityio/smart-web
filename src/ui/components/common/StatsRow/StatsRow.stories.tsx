import type { Meta, StoryObj } from '@storybook/react';
import { StatsRow } from '@/ui/components/common/StatsRow';

const meta: Meta<typeof StatsRow> = {
  component: StatsRow,
  tags: ['autodocs'],
  parameters: {
    layout: 'centered'
  },
  argTypes: {
    views: {
      description: 'Number of views',
      control: { type: 'number' }
    },
    favorite: {
      description: 'Number of favorites',
      control: { type: 'number' }
    },
    comment: {
      description: 'Number of comments',
      control: { type: 'number' }
    },
    isColumn: {
      description: 'Display in column layout',
      control: { type: 'boolean' }
    },
    viewIcon: {
      description: 'Icon for views',
      control: { type: 'text' }
    },
    favoriteIcon: {
      description: 'Icon for favorites',
      control: { type: 'text' }
    },
    commentIcon: {
      description: 'Icon for comments',
      control: { type: 'text' }
    }
  }
};

export default meta;

type Story = StoryObj<typeof StatsRow>;

export const Default: Story = {
  args: {
    views: 120,
    favorite: 30,
    comment: 15,
    isColumn: false,
    viewIcon: '/view-dark.svg',
    favoriteIcon: '/likes.svg',
    commentIcon: '/comments.svg'
  }
};
