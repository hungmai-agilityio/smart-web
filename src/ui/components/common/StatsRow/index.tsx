import { clsx } from 'clsx';

//Components
import { Photo } from '@/ui/components/common';

interface StatsRowProps {
  views: number;
  favorite: number;
  comment: number;
  isColumn?: boolean;
  viewIcon: string;
  favoriteIcon: string;
  commentIcon: string;
}

export const StatsRow = ({
  views,
  favorite,
  comment,
  isColumn,
  viewIcon,
  favoriteIcon,
  commentIcon
}: StatsRowProps) => {
  return (
    <div
      className={clsx('flex px-3 justify-between md:items-center w-full', {
        'flex-col md:hidden gap-2': isColumn
      })}
    >
      <div className="flex gap-1 items-center">
        <div
          className={clsx('w-icon-xs h-icon-xxs', {
            'w-7 h-icon-sm': isColumn
          })}
        >
          <Photo src={viewIcon} alt="view icon" autoSize />
        </div>
        <p
          className={clsx(
            'xl:w-10 w-8 truncate overflow-hidden',
            isColumn && 'w-full text-white'
          )}
        >
          {views}
        </p>
      </div>

      <div className="flex gap-1 items-center">
        <div
          className={clsx('w-icon-xxs h-icon-xxs', {
            'w-5 h-icon-xs': isColumn
          })}
        >
          <Photo src={favoriteIcon} alt="likes icon" autoSize />
        </div>
        <p
          className={clsx(
            'xl:w-10 w-8 truncate overflow-hidden',
            isColumn && 'w-full text-white'
          )}
        >
          {favorite}
        </p>
      </div>

      <div className="flex gap-1 items-center">
        <div
          className={clsx('w-icon-xxs h-icon-xxs', {
            'w-5 h-icon-xs': isColumn
          })}
        >
          <Photo src={commentIcon} alt="comments icon" autoSize />
        </div>
        <p
          className={clsx(
            'xl:w-10 w-8 truncate overflow-hidden',
            isColumn && 'w-full text-white'
          )}
        >
          {comment}
        </p>
      </div>
    </div>
  );
};
