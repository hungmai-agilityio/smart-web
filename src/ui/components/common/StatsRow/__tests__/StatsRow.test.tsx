import { StatsRow } from '@/ui/components/common/StatsRow';
import { render, screen } from '@testing-library/react';

describe('StatsRow component', () => {
  const defaultProps = {
    views: 12345,
    favorite: 1111,
    comment: 5432,
    viewIcon: '/view-dark.svg',
    favoriteIcon: '/likes.svg',
    commentIcon: '/comments.svg'
  };

  it('Should render default props', () => {
    render(<StatsRow {...defaultProps} />);

    const viewIcon = screen.getByAltText('view icon');
    const likeIcon = screen.getByAltText('likes icon');
    const commentIcon = screen.getByAltText('comments icon');
    expect(viewIcon).toBeInTheDocument();
    expect(likeIcon).toBeInTheDocument();
    expect(commentIcon).toBeInTheDocument();
  });

  it('should be render match to snapshot', () => {
    const { container } = render(<StatsRow {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });
});
