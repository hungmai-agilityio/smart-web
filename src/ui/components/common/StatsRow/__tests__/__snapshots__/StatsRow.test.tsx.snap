// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatsRow component should be render match to snapshot 1`] = `
<div>
  <div
    class="flex px-3 justify-between md:items-center w-full"
  >
    <div
      class="flex gap-1 items-center"
    >
      <div
        class="w-icon-xs h-icon-xxs"
      >
        <div
          class="relative w-full h-full"
        >
          <img
            alt="view icon"
            class="transition-opacity duration-300"
            data-nimg="fill"
            decoding="async"
            loading="lazy"
            src="/view-dark.svg"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent;"
          />
        </div>
      </div>
      <p
        class="xl:w-10 w-8 truncate overflow-hidden"
      >
        12345
      </p>
    </div>
    <div
      class="flex gap-1 items-center"
    >
      <div
        class="w-icon-xxs h-icon-xxs"
      >
        <div
          class="relative w-full h-full"
        >
          <img
            alt="likes icon"
            class="transition-opacity duration-300"
            data-nimg="fill"
            decoding="async"
            loading="lazy"
            src="/likes.svg"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent;"
          />
        </div>
      </div>
      <p
        class="xl:w-10 w-8 truncate overflow-hidden"
      >
        1111
      </p>
    </div>
    <div
      class="flex gap-1 items-center"
    >
      <div
        class="w-icon-xxs h-icon-xxs"
      >
        <div
          class="relative w-full h-full"
        >
          <img
            alt="comments icon"
            class="transition-opacity duration-300"
            data-nimg="fill"
            decoding="async"
            loading="lazy"
            src="/comments.svg"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent;"
          />
        </div>
      </div>
      <p
        class="xl:w-10 w-8 truncate overflow-hidden"
      >
        5432
      </p>
    </div>
  </div>
</div>
`;
